src/python/pants/backend/python/rules/pex.py

```diff
@@ -332,8 +304,10 @@ async def create_pex(
     else:
         argv.extend(request.interpreter_constraints.generate_pex_arg_list())
 
-    pex_debug = PexDebug(log_level)
-    argv.extend(pex_debug.iter_pex_args())
+    argv.append("--no-emit-warnings")
```

Comment:
This is controlled separately, e.g. a dedicated option on `python_binary` targets. We don't want to unconditionally set this.

Reply:
It was previously set unconditionally if logging was enabled. What is the effect of not setting it?

Reply:
Oh, I see. I'm not sure why it was originally there. @jsirois do you remember why you added that line?

Reply:
Any warning emitted by pex about internal pexes used by Pants would be noise to end users. For example the warning emitted when the Pex runtime detects old style namespace packages (`__import__('pkg_resources').declare_namespace(__name__)`) in a distribution used in the PEX that have no corresponding dependency declared on `setuptools`: https://github.com/pantsbuild/pex/blob/b51b340bea8a585a246997071a815d008986d6e3/pex/environment.py#L396-L400

Reply:
And the fact these pex binaries are incedental to the user (they're for pants only) vs the binaries we generate for users direct consumption for `python_binary` targets drives this difference in `--emit-warnings` configuration.

Reply:
Thank you for the explanation.

> vs the binaries we generate for users direct consumption

This is a bug now. Right now, we are unconditionally setting `--no-emit-warnings`. `python_create_binary.py` will optionally add `--no-emit-warnings` again to `PexRequest.additional_args` if relevant, which would result in the flag showing up twice. In `create_python_binary.py`, we need to be able to tell the `PexRequest` to not automatically set the flag.

I'll fix this in a followup PR.